### PR TITLE
Add extractLinksAsTexts and safety update extractLinks.

### PR DESCRIPTION
**Pull request by kaching88**

### CODE REVIEW

**Comment by kaching88** [comment]
I adapt code to your sugestions. Those misleads with javadoc was bring on because javadoc preview in Eclipse acting differently than true javadoc seen in browser. Thanks for your advices.

File: src/main/java/org/nibor/autolink/LinkExtractor.java [hunk 1]
```diff
@@ -27,11 +27,12 @@ public static Builder builder() {
...
     /**
      * Extract the links from the input text. Can be called multiple times with different inputs (thread-safe).
-     *
-     * @param input the input text, must not be {@code null}
+     * @param input the input text
      * @return a lazy iterable for the links in order that they appear in the input, never {@code null}
+     * @throws NullPointerException if the input argument is null 
      */
     public Iterable<LinkSpan> extractLinks(final CharSequence input) {
+    	Objects.requireNonNull(input);
```
Comment (robinst) [inline src/main/java/org/nibor/autolink/LinkExtractor.java]: Could you add a message as a second argument, like this?:

    Objects.requireNonNull(input, "input must not be null");

File: src/main/java/org/nibor/autolink/LinkExtractor.java [hunk 2]
```diff
@@ -39,6 +40,22 @@ public static Builder builder() {
...
         };
     }
+    
+    /**
+     * Extracts all links from input text to list of strings.
+     * @param input the input text
+     * @return an list containing all of the links from input
+     * @throws NullPointerException if the input argument is null
+     */
+    public List<String> extractLinksAsTexts(CharSequence input) {
```
Comment (robinst) [inline src/main/java/org/nibor/autolink/LinkExtractor.java]: I think `extractLinkTexts` or `extractLinksAsText` would be the better name.

File: src/test/java/org/nibor/autolink/LinkExtractorIterableTest.java [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+import static org.junit.Assert.assertTrue;
 
 import java.util.Iterator;
+import java.util.List;
 import java.util.NoSuchElementException;
 
-import static org.junit.Assert.*;
+import org.junit.Test;
 
 public class LinkExtractorIterableTest {
```
Comment (robinst) [inline src/test/java/org/nibor/autolink/LinkExtractorIterableTest.java]: Until now, this test was only about the iterable part. Could you rename it to `LinkExtractorTest`?

**Comment by robinst** [comment]
See inline comments, rest looks good now.

By the way, there is no need to create a new pull request, update the existing one instead. You can either amend your original commit and push it to the same branch, or add a new commit with the changes on top and push.

**Comment by robinst** [comment]
Hey. Looks like I didn't get a notification about your new commits. Unfortunately, your code doesn't compile anymore (you only renamed the test class, not the file). Could you fix that?

Also, the indentation should use 4 spaces. You used tabs in some places. May I ask which IDE/editor you are using? Maybe I can add a config file that enforces the right style.

**Pull request closed without merging**